tests/Pattern.elm

```diff
@@ -73,15 +73,42 @@ constructor =
         [ test "simple" <|
             \() ->
                 "A b c"
-                    |> isPattern (PConstructor "A" [ PVariable "b", PVariable "c" ])
+                    |> isPattern
+                        (PApplication
+                            (PApplication (PConstructor "A") (PVariable "b"))
+                            (PVariable "c")
+                        )
         , test "literal" <|
             \() ->
                 "A 1 2.5 'a' \"b\" _ c"
-                    |> isPattern (PConstructor "A" [ integerPattern 1, floatPattern 2.5, characterPattern 'a', stringPattern "b", PWild, PVariable "c" ])
+                    |> isPattern
+                        (PApplication
+                            (PApplication
+                                (PApplication
+                                    (PApplication
+                                        (PApplication
+                                            (PApplication (PConstructor "A") (integerPattern 1))
+                                            (floatPattern 2.5)
+                                        )
+                                        (characterPattern 'a')
+                                    )
+                                    (stringPattern "b")
+                                )
+                                PWild
+                            )
+                            (PVariable "c")
+                        )
```

Comment:
Can be done much nicer with fold like here:
https://github.com/Bogdanp/elm-ast/blob/0e8caebf225328ee0c93d3d8bb0ba354f4dfbc43/tests/Helpers.elm#L369